burn-common/src/id.rs

```diff
@@ -0,0 +1,78 @@
+use alloc::string::{String, ToString};
+
+use crate::rand::{get_seeded_rng, Rng, SEED};
+
+use uuid::{Builder, Bytes};
+
+pub struct IdGenerator {}
+
+impl IdGenerator {
+    pub fn generate() -> String {
+        let mut seed = SEED.lock().unwrap();
+        let mut rng = if let Some(rng_seeded) = seed.as_ref() {
+            rng_seeded.clone()
+        } else {
+            get_seeded_rng()
+        };
+
+        let random_bytes: Bytes = rng.gen();
+        *seed = Some(rng);
+
+        let uuid = Builder::from_random_bytes(random_bytes).into_uuid();
+
+        uuid.as_hyphenated().to_string()
```

Comment:
I'm not sure why we should use a seed here, is it to be `no_std` compatible?

Reply:
nanoid() was pulling random numbers from OS's getrandom directly which always gives a fresh random numbers. It's like getting from_entropy() each time. Since no_std does not have access to getrandom, we use our pre-seeding. It's similar what you wrote for tensor.random(). nanoid() was a big sticking point when I was building for no_std targets. Plus this approach after because of amortization for subsequent ID generation. 

I am using UUID's v4 with random data. It's good for database ids. Any other version relies on time and globally unique sequence generator (MAC id). For our use case V4 is sufficient and on par with nanoid.